src/Apps/Legion/Contexts/AppDataContext/AppDataContext.types.ts

```diff
@@ -1,15 +1,19 @@
+import { IDataManagementAdapter } from '../../Adapters/Standalone/DataManagement/Models/DataManagementAdapter.types';
+import { IADXAdapterTargetContext } from '../../Models/Interfaces';
+
 export interface ITargetDatabaseConnection {
     clusterUrl: string;
     databaseName: string;
 }
 export interface IAppDataContextProviderProps {
+    adapter?: IDataManagementAdapter;
```

Comment:
is it really optional? seems like if we need to fetch data we should juts make someone provide it.

Reply:
if you dont provide an adapter MockAdapter is going to be used through useADXAdapter hook

Reply:
right. I just kinda feel like the props should be designed for a real implmentation vs designing for mock data.